### PR TITLE
Fix: #144 Web support implementation

### PR DESCRIPTION
**Pull request by Sharcoux**
Signed-off-by: François BILLIOUD <f.billioud@gmail.com>

Summary:
---------
Making the Slider compatible with React-Native-Web and fixing #144 

Test Plan:
----------
1) change the package name for @yourname/slider
2) npm publish
3) On a react-native-web project do npm i @yourname/slider
4) import the slider and use it

### CODE REVIEW

**Comment by Sharcoux** [comment]
You can also test by using directly `npm i -S @sharcoux/slider`

**Comment by michalchudziak** [comment]
Great job @Sharcoux ! I'll try to test this functionality ASAP and release in the next version :)

**Comment by michalchudziak** [comment]
Hey, I've created a snack to test your implementation https://snack.expo.io/@michalchudziak/rebellious-beef-jerky and the slider is unmoveable

**Comment by Sharcoux** [comment]
> Hey, I've created a snack to test your implementation https://snack.expo.io/@michalchudziak/rebellious-beef-jerky and the slider is unmoveable

This is because the step is 1 by default... and you set the max value to 1

**Comment by Sharcoux** [comment]
Hum, wait a minute. It works if I set the step, but not if I don't set it. Let me check and fix if needed.

**Comment by Sharcoux** [comment]
Ok @michalchudziak , this is because of the defaultProps being 0 for step in the Slider.js
Why is that by the way? What does it mean and how is it handled on mobile?

**Comment by michalchudziak** [comment]
Cool, how do you feel about cherry-picking and resolving conflicts from https://github.com/react-native-community/react-native-slider/pull/81 - it seems to be really good implementation and I'd like to credit you both :)

**Comment by michalchudziak** [comment]
@Sharcoux I think it uses native default step

**Comment by Sharcoux** [comment]
> @Sharcoux I think it uses native default step

@michalchudziak nope. I just checked and it's using this code when step is not set:

```
    return Math.max(
      this.props.minimumValue,
      Math.min(
        this.props.maximumValue,
        ratio * (this.props.maximumValue - this.props.minimumValue) +
          this.props.minimumValue,
      ),
    );
```
I think it would be too complicated to cherry-pick from the other PR, but I can implement such a 'no step provided' behaviour. Give me 10 minutes

**Comment by Sharcoux** [comment]
> Cool, how do you feel about cherry-picking and resolving conflicts from #81 - it seems to be really good implementation and I'd like to credit you both :)

Which one seem to be a good implementation? Because I have serious doubts on #81 implementation. Especially for performances.

**Comment by Sharcoux** [comment]
Ok @michalchudziak you can test on your snack by upgrading to 2.0.11 and see that it's fine now.

**Comment by Sharcoux** [comment]
Is the current code ok for you @michalchudziak ?
I'm working on making it accessible as mentionned by necolas, but I'd like to ensure that everything is fine beside that.

**Comment by Sharcoux** [comment]
@michalchudziak , I added the accessibility. Do you have any remarks? Can we merge?

**Comment by michalchudziak** [comment]
@Sharcoux I'll check it out tomorrow, meanwhile can you fix the CI?

**Comment by Sharcoux** [comment]
> @Sharcoux I'll check it out tomorrow, meanwhile could you fix the CI  ?

Yeah, I'm sorry, I was fixing and testing some stuff for accessibility. I'm done now. The slider is accessible and can be controlled with arrow keys. You can try out the @sharcoux/slider version 2.0.15
or here : https://snack.expo.io/MqyqPrdMT

**Comment by Sharcoux** [comment]
@michalchudziak is it ok for merging now?

**Comment by michalchudziak** [comment]
@Sharcoux Yes, I think we are good to go!

**Comment by elisechant** [comment]
Where is the documentation for how this can be used?

**Comment by Sharcoux** [comment]
This is used the same way as for react native. Do you think that something is missing?
It is true that there are some props for the Web slider that don't exist on the rn slider. But I think that it should be documented once introduced in the native slider too. There is a feature request about this actually.

**Comment by elisechant** [comment]
@Sharcoux when using this package as a universal component, from my test, the web component can't be simply exchanged for the native component like usual via webpack alias: 

```
config.resolve.alias = {
  '@react-native-community/slider': '@react-native-community/slider/src/js/RNCSliderNativeComponent.web.js'
}
```

I would expect that alias mapping to resolve in a react-native-web web instance.

**Comment by Sharcoux** [comment]
You don't need to alias anything. Everything should be working.

File: src/js/RNCSliderNativeComponent.web.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    // const width = (containerSize.current ? containerSize.current.width : 0)
+    // const valueOffset = (inverted ? (1 - percentageValue) : percentageValue) * width
+
+    const thumbViewStyle = StyleSheet.compose(
+      {
+        width: thumbSize,
+        height: thumbSize,
+        // left: valueOffset - thumbSize / 2,
+        // top: trackHeight / 2 - thumbSize / 2,
+        // position: absolute,
```
Comment (daxfrost) [inline src/js/RNCSliderNativeComponent.web.js:215]: I think this should be absolute? it seems to cause a bug for the web... the button width is pushing the maximumTrackStyle view to the right and causing the width of the slider to be rendered incorrectly.

My container is pink and see the slider rendering at 100% width, but outside the cointainer:

image

Now, I am going to delete the DOM element for the thumbView in the Chrome inspector:

image

Comment (Sharcoux) [reply src/js/RNCSliderNativeComponent.web.js:215]: You are right and this was fixed later in a pr that was merged yesterday

Comment (daxfrost) [reply src/js/RNCSliderNativeComponent.web.js:215]: 🙌🏻 wonderful! This is an awesome addition, thank you for doing it

**Comment by prettyflvcko** [comment]
hey there, 

is this also available for ios or just for web? As I understand it, it's only available for web..

**Comment by Sharcoux** [comment]
So, at the moment, we have react-native-community/react-native-slider that has a different implementation for android/iOS/web, and @sharcoux/slider that has a single implementation for all platforms. But both projects are working on all environments.

**Pull request merged**